src/solutions/solution_interface.jl

```diff
@@ -1,7 +1,3 @@
-### Abstract Interface
-struct AllObserved
-end
```

Comment:
What's the replacement for this? And this would be technically breaking.

Reply:
Oh, moved to https://github.com/SciML/RecursiveArrayTools.jl/pull/143 ?

Reply:
https://github.com/SciML/RecursiveArrayTools.jl/pull/143 moves this upstream to  https://github.com/SciML/RecursiveArrayTools.jl/blob/5f02cb8c603c0d854cbcf8d148720ea503383a8d/src/vector_of_array.jl#L19. This is needed for `issymbollike(x)` function which is in turn needed for symbol resolution in `Base.getindex(...)`. 

(This is reexported by SciMLBase.jl)

Reply:
Yeah, we really need a SymbolicsBase.jl which just defines a few functions and types to use instead of that `issymbollike(x)` hack. @YingboMa is now the time?

Reply:
We can use https://github.com/JuliaSymbolics/TermInterface.jl